src/containers/org/components/SelectRoles.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+import React, { useState } from 'react';
+
+import debounce from 'lodash/debounce';
+import { List, Map } from 'immutable';
+import { Types } from 'lattice';
+import {
+  // $FlowFixMe[missing-export]
+  List as LUKList,
+  SearchInput,
+  Typography,
+} from 'lattice-ui-kit';
+import { ReduxUtils, useRequestState } from 'lattice-utils';
+import { useSelector } from 'react-redux';
+import type { Role } from 'lattice';
+
+import RoleListItem from './RoleListItem';
+
+import { Spinner } from '../../../components';
+import { GET_CURRENT_ROLE_AUTHORIZATIONS } from '../../../core/permissions/actions';
+import { PERMISSIONS } from '../../../core/redux/constants';
+import { selectCurrentRoleAuthorizations } from '../../../core/redux/selectors';
+import { getUserProfile } from '../../../utils';
+
+const { isSuccess } = ReduxUtils;
+const { PermissionTypes } = Types;
+
+type Props = {
+  members :Map;
+  onClick :(role :Role) => void;
+  roles :Role[];
+  selectedRoles :Map;
+};
+
+const SelectRoles = ({
+  members,
+  onClick,
+  roles,
+  selectedRoles,
+} :Props) => {
+
+  const currentRoleAuthorizations :Map = useSelector(selectCurrentRoleAuthorizations());
+  const requestState = useRequestState([PERMISSIONS, GET_CURRENT_ROLE_AUTHORIZATIONS]);
+  const [filterQuery, setFilterQuery] = useState('');
+
+  const success = isSuccess(requestState);
+  const debounceSetSearchTerm = debounce((value) => {
+    setFilterQuery(value);
+  }, 250);
+
+  const onSearchInputChange = (event :SyntheticEvent<HTMLInputElement>) => {
+    debounceSetSearchTerm(event.currentTarget.value);
+  };
+
+  let filteredRoles = roles;
+  if (filterQuery) {
+    filteredRoles = filteredRoles
+      .filter((role) => role.title.toLowerCase().includes(filterQuery.toLowerCase()));
+  }
+
+  const { name, email } = getUserProfile(members.first());
+  const selectedText = members.size === 1
+    ? (name || email)
+    : `${members.size} users`;
+
+  return (
+    <div>
+      <Typography color="textSecondary" gutterBottom>
+        {`These are all the roles you can add to ${selectedText}.`}
+      </Typography>
+      <SearchInput
+          onChange={onSearchInputChange}
+          placeholder="Search for a role by name" />
+      <LUKList>
+        { !success && <Spinner size="3x" />}
```

Comment:
so if this fails, will it just continue to spin?

Reply:
We can fix forward on this. no designs on error messages and such. if the request fails, then we do not know which roles the user is authorized to assign.

Reply:
okkkk